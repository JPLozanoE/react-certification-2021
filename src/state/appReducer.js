import { types } from '../types/types';

export const appReducer = (state = {}, action) => {
  switch (action?.type) {
    case types.changeSearchQuery:
      return {
        ...state,
        search: action.payload,
      };

    case types.setSelectedVideo:
      return {
        ...state,
        selectedVideo: action.payload,
      };

    case types.toggleTheme:
      return {
        ...state,
        isDarkTheme: !state.isDarkTheme,
      };

    case types.setRecommendedVideos:
      return {
        ...state,
        videos: action.payload,
      };

    case types.login:
      return {
        ...state,
        auth: {
          isLogged: true,
          displayName: action.payload.displayName,
          uid: action.payload.uid,
        },
      };

    case types.logout:
      return {
        ...state,
        auth: {
          isLogged: false,
          displayName: '',
          uid: null,
        },
      };

    case types.addFavoriteVideo:
      return {
        ...state,
        favoriteVideos: [
          ...state.favoriteVideos,
          { ...action.payload, isFavorite: true },
        ],
      };

    case types.deleteFavoriteVideo:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: false } : video
        ),
        favoriteVideos: state.favoriteVideos.filter(
          (favoriteVideo) => favoriteVideo.id.videoId !== action.payload
        ),
      };

    case types.updateRecommendedVideos:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: true } : video
        ),
      };

    default:
      return state;
  }
};
