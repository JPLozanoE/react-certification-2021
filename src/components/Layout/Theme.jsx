import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { AppContext } from '../../state/AppContext';

const Theme = ({ children }) => {
  const {
    state: { isDarkTheme },
  } = useContext(AppContext);
  const theme = {
    main: {
      ...(isDarkTheme ? { backgroundColor: '#181818' } : { backgroundColor: '#ffffff' }),
    },
    titles: {
      ...(isDarkTheme ? { color: '#ffffff' } : { color: '#00000' }),
    },
    subtitles: {
      ...(isDarkTheme ? { color: 'rgb(170,170,170)' } : { color: 'rgb(96,96,96)' }),
    },
  };

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default Theme;
