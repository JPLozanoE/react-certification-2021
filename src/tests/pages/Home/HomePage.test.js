/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { AppContext } from '../../../state/AppContext';
import Theme from '../../../components/Layout/Theme';
import { demoState } from '../../fixtures/demoState';
import HomePage from '../../../pages/Home/Home.page';

describe('Home Page', () => {
  let wrapper;

  test('Should render properly', () => {
    const providerValues = {
      state: { ...demoState },
      dispatch: jest.fn(),
    };
    wrapper = render(
      <AppContext.Provider value={providerValues}>
        <Theme>
          <Router>
            <HomePage />
          </Router>
        </Theme>
      </AppContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper.getByText(demoState.videos[0].snippet.channelTitle)
    ).toBeInTheDocument();
  });
});
