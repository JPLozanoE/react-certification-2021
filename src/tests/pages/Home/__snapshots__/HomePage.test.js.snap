// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM gBYQnL"
      >
        <div
          class="sc-gtsrHT eMPrOS"
        >
          <img
            alt="img"
            aria-label="Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights"
            height="150px"
            src="https://i.ytimg.com/vi/yFq-c9ZpmRs/hqdefault.jpg"
            style="object-fit: cover;"
            width="100%"
          />
          <div
            style="padding: 10px;"
          >
            <div>
              <h3
                class="sc-dlnjwi dzceGW"
              >
                Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights
              </h3>
            </div>
            <p
              class="sc-hKFxyN cvlDaB"
            >
              MLB
            </p>
            <p
              class="sc-hKFxyN cvlDaB"
            >
              Publicado el 
              2021-08-21T03:02:50Z
            </p>
          </div>
        </div>
        <div
          class="sc-gtsrHT eMPrOS"
        >
          <img
            alt="img"
            aria-label="State of Emergency As Hurricane Henri Approaches New York"
            height="150px"
            src="https://i.ytimg.com/vi/EqG9gWbwxlw/hqdefault.jpg"
            style="object-fit: cover;"
            width="100%"
          />
          <div
            style="padding: 10px;"
          >
            <div>
              <h3
                class="sc-dlnjwi dzceGW"
              >
                State of Emergency As Hurricane Henri Approaches New York
              </h3>
            </div>
            <p
              class="sc-hKFxyN cvlDaB"
            >
              NBC News
            </p>
            <p
              class="sc-hKFxyN cvlDaB"
            >
              Publicado el 
              2021-08-21T23:30:18Z
            </p>
          </div>
        </div>
        <div
          class="sc-gtsrHT eMPrOS"
        >
          <img
            alt="img"
            aria-label="MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!"
            height="150px"
            src="https://i.ytimg.com/vi/LqiDJh4zxAc/hqdefault.jpg"
            style="object-fit: cover;"
            width="100%"
          />
          <div
            style="padding: 10px;"
          >
            <div>
              <h3
                class="sc-dlnjwi dzceGW"
              >
                MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!
              </h3>
            </div>
            <p
              class="sc-hKFxyN cvlDaB"
            >
              Magic Secrets Revealed
            </p>
            <p
              class="sc-hKFxyN cvlDaB"
            >
              Publicado el 
              2019-09-05T21:00:03Z
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM gBYQnL"
    >
      <div
        class="sc-gtsrHT eMPrOS"
      >
        <img
          alt="img"
          aria-label="Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights"
          height="150px"
          src="https://i.ytimg.com/vi/yFq-c9ZpmRs/hqdefault.jpg"
          style="object-fit: cover;"
          width="100%"
        />
        <div
          style="padding: 10px;"
        >
          <div>
            <h3
              class="sc-dlnjwi dzceGW"
            >
              Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights
            </h3>
          </div>
          <p
            class="sc-hKFxyN cvlDaB"
          >
            MLB
          </p>
          <p
            class="sc-hKFxyN cvlDaB"
          >
            Publicado el 
            2021-08-21T03:02:50Z
          </p>
        </div>
      </div>
      <div
        class="sc-gtsrHT eMPrOS"
      >
        <img
          alt="img"
          aria-label="State of Emergency As Hurricane Henri Approaches New York"
          height="150px"
          src="https://i.ytimg.com/vi/EqG9gWbwxlw/hqdefault.jpg"
          style="object-fit: cover;"
          width="100%"
        />
        <div
          style="padding: 10px;"
        >
          <div>
            <h3
              class="sc-dlnjwi dzceGW"
            >
              State of Emergency As Hurricane Henri Approaches New York
            </h3>
          </div>
          <p
            class="sc-hKFxyN cvlDaB"
          >
            NBC News
          </p>
          <p
            class="sc-hKFxyN cvlDaB"
          >
            Publicado el 
            2021-08-21T23:30:18Z
          </p>
        </div>
      </div>
      <div
        class="sc-gtsrHT eMPrOS"
      >
        <img
          alt="img"
          aria-label="MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!"
          height="150px"
          src="https://i.ytimg.com/vi/LqiDJh4zxAc/hqdefault.jpg"
          style="object-fit: cover;"
          width="100%"
        />
        <div
          style="padding: 10px;"
        >
          <div>
            <h3
              class="sc-dlnjwi dzceGW"
            >
              MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!
            </h3>
          </div>
          <p
            class="sc-hKFxyN cvlDaB"
          >
            Magic Secrets Revealed
          </p>
          <p
            class="sc-hKFxyN cvlDaB"
          >
            Publicado el 
            2019-09-05T21:00:03Z
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
